### PR TITLE
server: Remove periodic heap profiling

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Ref #15927 

What's Changed:

```commit-message
Remove periodic heap profiling
```
As heap profiling is already enabled by default, there is no need for periodic heap profiling.

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Remove periodic heap profiling
```

### CODE REVIEW

File: src/server/status_server/mod.rs [hunk 1]
```diff
@@ -651,20 +651,20 @@ where
                             (Method::GET, "/status") => Ok(Response::default()),
                             (Method::GET, "/debug/pprof/heap_list") => {
                                 return Ok(make_response(
-                                    StatusCode::NOT_IMPLEMENTED,
-                                    "deprecated",
+                                    StatusCode::NOT_FOUND,
```
Comment (LykxSassinator) [inline src/server/status_server/mod.rs:654]: IMO, it's better to use "METHOD_NOT_ALLOWED" or "GONE" to notify the caller that these methods are deprecated.

**Review by LykxSassinator — COMMENTED** [review]
Some clippy errors should be fixed, others LGTM.

File: src/server/status_server/mod.rs [hunk 2]
```diff
@@ -749,12 +649,23 @@ where
                                 Self::handle_get_metrics(req, &cfg_controller)
                             }
                             (Method::GET, "/status") => Ok(Response::default()),
-                            (Method::GET, "/debug/pprof/heap_list") => Self::list_heap_prof(req),
+                            (Method::GET, "/debug/pprof/heap_list") => {
+                                return Ok(make_response(
```
Comment (LykxSassinator) [inline src/server/status_server/mod.rs:653]:
```suggestion
                                Ok(make_response(
```

**Review by glorv — APPROVED** [review]

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**